x-pack/plugins/observability_solution/infra/server/plugin.ts

```diff
@@ -192,6 +198,22 @@ export class InfraServerPlugin
       { sources }
     );
 
+    const alertsLocator = plugins.share.url.locators.create(new AlertsLocatorDefinition());
```

Comment:
the  server side `alertLocator` is already created here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability/server/plugin.ts#L101. 

So on this line, we could just retrieve its reference

```suggestion
    const alertsLocator = plugins.share.url.locators.get<AlertsLocatorParams>(alertsLocatorID);
```